### PR TITLE
Add system config htaccess.IgnoreFrontController for prettyURLs w/o mod_env

### PR DESCRIPTION
**Pull request by eppfel**
As discussed with @LukasReschke at nextcloud conf.

Mabye we need some tests in lib/UrlGeneratorTest.php?

### CODE REVIEW

**Comment by eppfel** [comment]
The conifg.example.php description could be improved, I guess:

For server setups, that don't have `mod_env` enabled or restricted (e.g. suEXEC)
 this parameter has to be set to true and will assume mod_rewrite.

 Please check, if `mod_rewrite` is active and functional before setting this
 parameter and you updated your .htaccess with `occ maintenance:update:htaccess`.
 Otherwise your nextcloud installation might not be reachable anymore.
 For example, try accessing resources by leaving out `index.php` in the URL.

**Comment by LukasReschke** [comment]
@eppfel Thanks! Can you sign the commit as described at https://github.com/nextcloud/server/blob/master/CONTRIBUTING.md#sign-your-work? Basically use `git commit -s` or in this case just a `git commit --amend -s` and then a force push to this branch :)

File: lib/private/URLGenerator.php [hunk 1]
```diff
@@ -93,7 +94,7 @@ public function linkToRouteAbsolute($routeName, $arguments = array()) {
 	 * Returns a url to the given app and file.
 	 */
 	public function linkTo( $app, $file, $args = array() ) {
-		$frontControllerActive = (getenv('front_controller_active') === 'true');
+		$frontControllerActive = ($this->config->getSystemValue('htaccess.IgnoreFrontController') === true || getenv('front_controller_active') === 'true');
```
Comment (nickvergessen) [inline lib/private/URLGenerator.php]: missing the default value.

**Review by nickvergessen — DISMISSED** [review]

**Comment by MorrisJobke** [comment]
@LukasReschke Is this the cause for the weird error message in the admin panel:

> php does not seem to be setup properly to query system environment variables. The test with getenv("PATH") only returns an empty response

Maybe we can make this an "info" level message instead of a red warning. Or what is the purpose of this message?

**Comment by LukasReschke** [comment]
> Maybe we can make this an "info" level message instead of a red warning. Or what is the purpose of this message?

Well. No idea. I didn't add it 😉 

=> https://github.com/owncloud/core/pull/15617

**Comment by MorrisJobke** [comment]
> => owncloud/core#15617

Okay - then simply leave it in.

**Comment by eppfel** [comment]
I rebased and had to move the changes in `core/js/config.php` to `core/tempate/JSConfigHelper.php`,  but I have no clue about the error drone threw.
@LukasReschke @nickvergessen Would you take a look?

**Comment by nickvergessen** [comment]
> error_outlineERROR: terminal inactive for 10m0s, build cancelled

Simple timeout, not your fault.

**Comment by eppfel** [comment]
Would be nice to get this in. Everything looks ready to merge.
@MorrisJobke @LukasReschke @nickvergessen ?

**Comment by nickvergessen** [comment]
> Conflicting files
lib/private/legacy/util.php

**Comment by eppfel** [comment]
Rebased.

**Comment by MorrisJobke** [comment]
Tested and works 👍 

@nickvergessen Could you review this please? Would be nice to get it in.

**Comment by MorrisJobke** [comment]
@rullzer @LukasReschke Please review :)

**Comment by LukasReschke** [comment]
LGTM

**Pull request merged**